File: yanhuo-xo/src/main/java/com/yanhuo/xo/entity/Comment.java
package com.yanhuo.xo.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.yanhuo.common.entity.BaseEntity;
import io.swagger.models.auth.In;
import lombok.Data;

/**
 * @author xiaozhao
 */
@Data
@TableName("t_comment")
public class Comment extends BaseEntity {

    private String nid;

    private String noteUid;

    private String uid;

    private String pid;

    private String replyId;

    private String replyUid;

    private Integer level;

    private Integer sort;

    private String content;

    private Long likeCount;

    private Long twoCommentCount;
}

File: yanhuo-web/src/type/comment.ts
export interface CommentDTO {
  nid: string;
  noteUid: string;
  pid: string;
  replyId: string;
  replyUid: string;
  replyUsername: string;
  level: number;
  content: string;
}

export interface Comment {
  id: string;
  pid: string;
  nid: string;
  noteCover: string;
  uid: string;
  username: string;
  avatar: string;
  replyId: string;
  replyUid: string;
  replyUsername: string;
  content: string;
  replyContent: string;
  time: number;
  likeCount: number;
  isLike: boolean;
  twoCommentCount: number;
  children: Array<Comment>;
}
